packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -171,6 +171,24 @@ export function matches(element: Element, selector: string | AST): boolean {
         pseudoClass.argument !== undefined &&
         matchCSSProperty(element, pseudoClass.argument, '::before')
       );
+    } else if (pseudoClass.name === 'xpath') {
+      const { argument } = pseudoClass;
+      if (argument === undefined) {
+        return false;
+      }
+
+      try {
+        // Create an XPath evaluator
+        const evaluator = new XPathEvaluator();
+        // Create an XPath expression
+        const expression = evaluator.createExpression(argument, null);
+        // Evaluate the expression
+        const result = expression.evaluate(element, XPathResult.BOOLEAN_TYPE, null);
```

Comment:
Does this tell us if `element` matches the xpath or if any of its descendants matches the xpath? It's a bit unclear to me reading the documentation but if it's the former, this seems like it would potentially apply too broadly. Shouldn't we instead iterate on all the nodes which match the xpath?